### ISSUE TITLE
iAP over BT - Data resumption

### ISSUE THREAD

**Issue by OlesiaV** [smartdevicelink/sdl_core#1661]
## Description:

_Currently SDL keeps data created by app in case of transport switching independenlty on value of `<hashID>`.
It may cause a lot of issues from mobile app perspective if app sends invalid "hashID" but SDL still keeps all data for this app.
As a result, to avoid all concerns -> SDL will keep data ONLY if app sends valid "hashID" during "reconnection_timer".
Otherwise, SDL will remove all data internally and trigger HMI to remove all data related to this app.
If mobile app re-registers after "reconnection_timer" -> SDL must behave per existing rules for Data Resumption._

## .ini file change ##

New parameters "AppTransportChangeTimer" , "AppTransportChangeTimerAddition"   must be added to .ini file -> [MAIN] section

```
[MAIN] 
"AppTransportChangeTimer" = 500 ms 
defines the timeout for waiting the mobile app to reconnect between Bluetooth and USB transports change

```

```
[MAIN] 
"AppTransportChangeTimerAddition" = 0ms
Defines the timeout  for waiting of every mobile app to reconnect between Bluetooth and USB transports change in case the number of connected apps is more than one
```

## Related Diagrams:

![iap_over_bt_data_resumption](https://user-images.githubusercontent.com/23719788/28126364-f09111d6-6731-11e7-8cf3-bb3bbd84c2ee.png)

## Links:

- **Requirement:**  [#37](https://github.com/smartdevicelink/sdl_requirements/issues/37) iAP over BT re-connection
- **Parent task:** [issues/1659](https://github.com/smartdevicelink/sdl_core/issues/1659)

**Comment by jacobkeeler** [comment]
@OlesiaV This is resolved with the merge of #1927, correct?

**Comment by jacobkeeler** [comment]
Related: #1587
